examples/unstructured_2d_dgsem/elixir_shallow_water_convergence.jl

```diff
@@ -0,0 +1,69 @@
+
+using Downloads: download
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# semidiscretization of the shallow water equations with a particular
+# bottom topography function
+
+@inline function mms_bottom_topography(x, equations::ShallowWaterEquations2D)
+  # periodic bottom for the periodic with a twist mesh domain [0, sqrt(2)]^2
+  return 2.0 + 0.5 * sin(sqrt(2.0)*pi*x[1]) + 0.5 * sin(sqrt(2.0)*pi*x[2])
+end
+
+equations = ShallowWaterEquations2D(9.81, bottom_topography=mms_bottom_topography)
+
+initial_condition = initial_condition_convergence_test
```

Comment:
This doesn't look good to me - if we use the ugly (but currently necessary) workaround of introducing additional conserved variables with vanishing fluxes for the bottom topography, the initial condition should know everything about the bottom topography. If you change either the bottom or the other part of the IC, you must also change the other one. Thus, splitting them doesn't seem right, I think.

Reply:
That is true. There is always this extra layer when you setup a shallow water problem that you need to pick the initial condition and a bottom topography that make sense together. That being said we can think about maybe adding another cache that dispatches on `ShallowWaterEquations2D`. This cache could contain the `bottom_topography` function as well as precompute and store the `b` values we need.

Reply:
Why don't we just set the bottom topography in the IC directly?

Reply:
Ah, I see what you mean. There is then no separate function. That could work too I suppose. I will think about this more once I get the discontinuous bottom stuff working.

Reply:
Discontinuous bottom topographies will probably require something as discussed in #814

Reply:
Yes it does. I have a hacky way of setting up such discontinuities with a modified `compute_coefficients!` routine.

Reply:
I removed the hacky way of setting the discontinuities using `compute_coefficients!` instead I modify `ode.u0` directly in the elixirs as suggested in the Trixi meeting.